### DISCUSSION TITLE
"Device busy" error even with '--force'. wsl not recognizing usbip

### DISCUSSION THREAD

**Discussion by adriencrater** [dorssel/usbipd-win#884] [Q&A]
I am trying to connect an x-box 360 controller to wsl2. I cannot figure out how to get rid of the "device busy" error. I have tried using --force and restarting the terminals. My wsl2 also does not recognize usbip or usbipd. I used the download wizard. Is there something else I need to do to connect my wsl2 as well? Sorry if I am missing something obvious, I'm new to programming.

**Comment by dorssel** [comment]
Device busy means that Windows (either software, a service, or the driver itself) has an open handle to the device; and therefore `usbipd-win` gets a "NO" from Windows when trying to claim it.

One solution is to find the software/service that is using it and stopping that. But often it is the driver itself that keeps the device open, so the driver cannot be replaced by the usbipd-win stub driver "on-the-fly".

--force does a permanent driver update, forcing the device to use the stub driver all the time, It usually requires a reboot.

After doing `usbipd bind --force`, do a reboot. After reboot do a `usbipd list`. Does the device show up as "Shared (forced)"? If so, then the driver switch worked. You should now be able to attach it to WSL.
